crates/net/network/src/config.rs

```diff
@@ -572,7 +572,7 @@ impl<N: NetworkPrimitives> NetworkConfigBuilder<N> {
     /// establishing a connection.
     pub fn build<C>(self, client: C) -> NetworkConfig<C, N>
     where
-        C: ChainSpecProvider<ChainSpec: Hardforks>,
+        C: ChainSpecProvider<ChainSpec: Hardforks + EthChainInitSpec>,
```

Comment:
Here it seems, the genesis file is not needed, its mostly the timestamp and difficulty parameters